api/services/contact_fields_mapping_service.py

```diff
@@ -0,0 +1,119 @@
+            user_details = models.User.query.get_by_phone(self.user_phone)
+            if not user_details:
+                return False
+
+            contact_groups = jsonData["contact"]["groups"]
+
+            if contact_groups:
+                for contact_group in contact_groups:
+                    contact_group_name = contact_group["name"]
+                    user_contact_group_mapping = (
+                        models.ContactFieldsMapping.query.get_by_group_name(
+                            contact_group_name
+                        )
+                    )
+                    if user_contact_group_mapping:
+                        self.process_contact_groups_data(
+                            user_contact_group_mapping, user_details
+                        )
+        except Exception as e:
+            print(
+                f"Error occurred while handling custom field mapping. Exception:  {e}"
+            )
+
+    def process_contact_fields_data(self, user_contact_field_details, user_details):
+        try:
+            for user_contact_field_detail in user_contact_field_details:
+                class_object = helpers.get_class_by_tablename(
+                    user_contact_field_detail.mapped_table_name
+                )
+                if class_object:
+                    column_name = user_contact_field_detail.mapped_table_column_name
+                    mapped_table_column_value = (
+                        user_contact_field_detail.mapped_table_column_value
+                    )
+                    self.update_mapped_fields(
+                        class_object,
+                        column_name,
+                        mapped_table_column_value,
+                        user_details,
+                    )
+        except Exception as e:
+            print(f"Exception occurred: {e}")
+
+    def process_contact_groups_data(self, user_contact_group_details, user_details):
+        try:
+            for user_contact_group_detail in user_contact_group_details:
+                class_object = helpers.get_class_by_tablename(
+                    user_contact_group_detail.mapped_table_name
+                )
+                if class_object:
+                    column_name = user_contact_group_detail.mapped_table_column_name
+                    mapped_table_column_value = (
+                        user_contact_group_detail.mapped_table_column_value
+                    )
+                    self.update_mapped_fields(
+                        class_object,
+                        column_name,
+                        mapped_table_column_value,
+                        user_details,
+                    )
+        except Exception as e:
+            print(f"Exception occurred: {e}")
+
+    def update_mapped_fields(
+        self, class_object, column_name, mapped_table_column_value, user_details
+    ):
+        try:
+            class_object_data = class_object.get_by_user_id(user_details.id)
```

Comment:
Which entry we will get in this object if the table is user program

Reply:
<api.models.user_program.UserProgram object at 0x0000024744ED4E20>